common/baselines/librispeech/ls960/gmm/baseline_args.py

```diff
@@ -0,0 +1,457 @@
+        "maximum_segment_length": 6000,
+        "iterations": 5,
+        "penalty": 0,
+        "minimum_speech_proportion": 0.7,
+        "save_alignment": False,
+        "keep_accumulators": False,
+        "extra_merge_args": None,
+        "extra_config": None,
+        "extra_post_config": None,
+    }
+
+    monophone_training_args = {
+        "name": "mono",
+        "feature_flow": feature_flow,
+        "feature_energy_flow_key": "energy," + feature_flow,
+        "align_iter": 75,
+        "splits": 10,
+        "accs_per_split": 2,
+        "dump_alignment_score_report": True,
+    }
+
+    monophone_recognition_args = {
+        # GmmSystem.recognition() args:
+        "iters": [10],
+        "lm_scales": [10],
+        "optimize_am_lm_scale": True,
+        # meta.System.recog() args:
+        "feature_flow": feature_flow,
+        "pronunciation_scales": [1.0],
+        "lm_lookahead": True,
+        "lookahead_options": None,
+        "create_lattice": True,
+        "eval_single_best": True,
+        "eval_best_in_lattice": True,
+        "search_parameters": {
+            "beam-pruning": 15.0,
+            "beam-pruning-limit": 100000,
+            "word-end-pruning": 0.5,
+            "word-end-pruning-limit": 10000,
+        },
+        "parallelize_conversion": False,
+        "lattice_to_ctm_kwargs": {
+            "fill_empty_segments": False,
+            "best_path_algo": "bellman-ford",
+        },
+        "rtf": 20,
+        "mem": 4,
+        "use_gpu": False,
+    }
+
+    return util.GmmMonophoneArgs(
+        linear_alignment_args, monophone_training_args, monophone_recognition_args
+    )
+
+
+def get_cart_args(
+    use_stress_marker: bool = False,
+    max_leaves: int = 12001,
+    min_obs: int = 1000,
+    hmm_states: int = 3,
+    feature_flow: str = "mfcc+deriv+norm",
+    add_unknown: bool = False,
+) -> util.GmmCartArgs:
+    """
+
+    :param use_stress_marker: use ARPAbet stress marker, please also check for correct lexicon then
+    :param max_leaves: total number of final CART states
+    :param min_obs: minimum observation count needed for a label
+    :param hmm_states: number of states per label for the HMM-topology
+    :param feature_flow:
+    :param add_unknown: set to true if an unknown phoneme exists
+    """
+
+    CartQuestions = (
+        CartQuestionsWithStress if use_stress_marker else CartQuestionsWithoutStress
+    )
+
+    cart_questions_class = CartQuestions(
+        max_leaves=max_leaves,
+        min_obs=min_obs,
+        add_unknown=add_unknown,
+    )
+
+    cart_questions = cart.PythonCartQuestions(
+        phonemes=cart_questions_class.phonemes_boundary_special,
+        steps=cart_questions_class.steps,
+        max_leaves=max_leaves,
+        hmm_states=hmm_states,
+    )
+
+    cart_lda_args = {
+        "name": "cart_mono",
+        "alignment": "train_mono",
+        "initial_flow_key": feature_flow,
+        "context_flow_key": feature_flow.split("+")[0],
```

Comment:
this only works by convention and not in general. But I'll allow it.

Reply:
I appreciate your generosity :)